commands/cluster-nodes.md

```diff
@@ -28,26 +28,53 @@ of output:
 824fe116063bc5fcf9f4ffd895bc17aee7731ac3 127.0.0.1:30006@31006 slave 292f8b365bb7edb5e285caf0b7e6ddc7265d2f4f 0 1426238317741 6 connected
 e7d1eecce10fd6bb5eb35b9f99a514335d9ba9ca 127.0.0.1:30001@31001 myself,master - 0 0 1 connected 0-5460
 ```
+Starting from 7.2.0, the output of the command always contains a new auxiliary
+field called shard-id. The following is an example of output on Redis 7.2.0.
+
+```
+07c37dfeb235213a872192d90877d0cd55635b91 127.0.0.1:30004@31004,,shard-id=69bc080733d1355567173199cff4a6a039a2f024 slave e7d1eecce10fd6bb5eb35b9f99a514335d9ba9ca 0 1426238317239 4 connected
+67ed2db8d677e59ec4a4cefb06858cf2a1a89fa1 127.0.0.1:30002@31002,,shard-id=114f6674a35b84949fe567f5dfd41415ee776261 master - 0 1426238316232 2 connected 5461-10922
+292f8b365bb7edb5e285caf0b7e6ddc7265d2f4f 127.0.0.1:30003@31003,,shard-id=fdb36c73e72dd027bc19811b7c219ef6e55c550e master - 0 1426238318243 3 connected 10923-16383
+6ec23923021cf3ffec47632106199cb7f496ce01 127.0.0.1:30005@31005,,shard-id=114f6674a35b84949fe567f5dfd41415ee776261 slave 67ed2db8d677e59ec4a4cefb06858cf2a1a89fa1 0 1426238316232 5 connected
+824fe116063bc5fcf9f4ffd895bc17aee7731ac3 127.0.0.1:30006@31006,,shard-id=fdb36c73e72dd027bc19811b7c219ef6e55c550e slave 292f8b365bb7edb5e285caf0b7e6ddc7265d2f4f 0 1426238317741 6 connected
+e7d1eecce10fd6bb5eb35b9f99a514335d9ba9ca 127.0.0.1:30001@31001,,shard-id=69bc080733d1355567173199cff4a6a039a2f024 myself,master - 0 0 1 connected 0-5460
+```
```

Comment:
Historically we just show the latest value and indicate what might be optional in past versions. I would just keep this latest value. We should also probably make it clear this format can change arbitrarily as is mostly textual.

Reply:
>We should also probably make it clear this format can change arbitrarily as is mostly textual.

This disclaimer seems a bit scary to me and I don't think it accurately captures our intent and efforts so far (to keep it as backward compatible as possible). 

Agreed with your other comments.

Reply:
> This disclaimer seems a bit scary to me and I don't think it accurately captures our intent and efforts so far (to keep it as backward compatible as possible).

Point taken. We might want to point out the proper way to "parse" it then, so we have some rules we hold as invariants.

Reply:
Would the following work? 

```
4. `[,auxiliary_field=value]*`: A list of comma-separated key-value pairs that
   represent various node properties, such as `shard-id`. There is no intrinsic
   order among auxiliary fields. The auxiliary fields can appear at
   different position in the list from release to release. Both the key name and
   value can contain ASCII alphanumeric characters and the characters in
   `!#$%&()*+-.:;<>?@[]^_{|}~` only. Auxiliary fields are explained in detail in
   the section below.
```

Frankly speaking, an air-tight definition could actually hurt extensibility. Imagine that we had asked everyone to parse the second column exactly as `ip:port@cport`, we would not be able to extend it to include the new fields like `hostname`, `shard-id`, etc. I think we have struck a good balance with the callout above on the new aux fields along with the existing rules but let me know if you think other invariants should be included too.

Reply:
I was actually going to just suggest that it is "space separated fields", the only invariant we plan on really holding is the lack of spaces. I still don't think end users should be parsing this information, and we've been pushing clients away from it.

Reply:
I don't know if it really hurts extensibility. One of the problems we had was we never told users any way to parse, and they were guessing at it. It also was inherently inextensible before, and now at least we have a mechanism.

Reply:
I suppose what I really still want to say is don't use this because it doesn't have a schema. Perhaps it's better to focus the efforts on the new API that provides admin information. So I'm okay leaving this, and we can mark it "deprecated" later.

Reply:
SGTM